### ISSUE TITLE
Comparison to other "safe transmute" crates

### ISSUE THREAD

**Issue by 8573** [GoldsteinE/cismute#1]
In the spirit of
* https://github.com/AltSysrq/proptest/issues/15#issuecomment-348382287,
* https://github.com/Marwes/combine/issues/73, and
* https://github.com/shepmaster/snafu/issues/50,

I suggest documenting how this crate compares with some other crates that offer "safe transmute" functions, especially for transmuting a type to itself, such as
* https://github.com/Centril/refl
* https://github.com/pthariensflame/identity.rs
* https://github.com/nabijaczleweli/safe-transmute-rs

**Comment by GoldsteinE** [comment]
Thanks for suggestion!

It looks like first two require to statically provide an equality proof, which could be impossible in some contexts — for example, when implementing a generic method of a foreign trait and arguably worse for usability: if you want to compare a type against several other types, you’ll need to carry an enum of `Refl`s and match on it dynamically, which is not zero-cost.

I think the third one doesn’t provide equivalent functionality at all, but I’ll have to look further into it.

**Comment by GoldsteinE** [comment]
I’ll describe this in README a little bit later.

**Comment by zicklag** [comment]
Another crate we might want to compare with is [castaway](https://docs.rs/castaway/latest/castaway/) which seems by far the most similar so far, but also seems to lack the functionality of `cismute::reference`.

**Comment by GoldsteinE** [comment]
It can cast simple references, but it seems to not be extendable. Also it uses macros instead of functions, so reqiured bounds are explained in text, not in types. It also lacks `switch`.

Thanks, I’ll add this to the comparison.

**Comment by zicklag** [comment]
Ah. I ended up using your crate instead of it because I didn't want to have to implement [`LifetimeFree`](https://docs.rs/castaway/latest/castaway/trait.LifetimeFree.html), especially because I had to implement it on a foreign container type, which meant I'd have to use a newtype, which would just end up with _way_ more boilerplate.

I think I have a real-world use-case for your crate, so thanks a lot for this!
